providers/universal-provider/src/UniversalProvider.ts

```diff
@@ -0,0 +1,267 @@
+import { EventEmitter } from "events";
+import SignClient from "@walletconnect/sign-client";
+import { ProviderAccounts } from "eip1193-provider";
+import { SessionTypes } from "@walletconnect/types";
+import { getSdkError } from "@walletconnect/utils";
+import { getDefaultLoggerOptions } from "@walletconnect/logger";
+import pino, { Logger } from "pino";
+import Eip155Provider from "./providers/eip155";
+import { getChainFromNamespaces } from "./utils";
+import {
+  IUniversalProvider,
+  IProvider,
+  RpcProviderMap,
+  ConnectParams,
+  RequestArguments,
+  UniversalProviderOpts,
+  NamespaceConfig,
+} from "./types";
+
+import { RELAY_URL, LOGGER, STORAGE } from "./constants";
+
+export class UniversalProvider implements IUniversalProvider {
+  public client!: SignClient;
```

Comment:
How come we need the non-null assertion operator here but didn't seem to previously in SignClient and other similar classes?

Reply:
Because the client is not initialized in the constructor and thus requiring adding `| undefinied`. 
This makes usage later on annoying due to constantly having to add checks such ass `client?.`

Thought this solution is fine as attempting to use the client without it being initialized results in an exception such as
https://github.com/WalletConnect/walletconnect-monorepo/blob/gancho/feat/universal-provider/providers/universal-provider/src/UniversalProvider.ts#L180

Reply:
Yeah makes total sense 👍 

I'm more surprised we didn't run into this before in e.g. SignClient since the pattern is the same, but it seems like the way the type referencing there refers back to the abstract class every time (i.e. `public session: ISignClient["session"]`) doesn't evoke this issue in the TS compiler